lib/spack/spack/bootstrap/status.py

```diff
@@ -161,7 +168,7 @@ def status_message(section):
     with ensure_bootstrap_configuration():
         missing_software = False
         for found, err_msg in required_software():
-            if not found:
+            if not found and err_msg:
```

Comment:
Is it ever possible for missing software to have an empty `err_msg`?

Reply:
Correct. Sometimes the best way to fix a mypy warning is to improve the code;

In the case, prevent an Exception which would normally not happen, but could happen if required software was changed in the future.

@tldahlgren Using mypy also means to think a bit different about about making code safe against failing.

Reply:
@bernhardkaindl So true!  Reminds me of discussions I've had many years ago about the flexibility of python (and writing "pythonic" code) versus (strong) type checking.

I am still learning how to instrument python to satisfy mypy, but I have noticed it is not quite as flexible as I'd hoped. Having previously worked with strongly typed languages and on dynamic and static analysis tools, I can understand why it makes some of the decisions it makes.  It definitely requires a shift in thinking about how to write python code.